tests/pr_test.go

```diff
@@ -39,7 +55,8 @@ func TestRunDefaultExample(t *testing.T) {
 }
 
 func TestRunUpgradeExample(t *testing.T) {
-	t.Parallel()
+	// DO NOT RUN MULTIPLE TESTS IN PARALLEL
```

Comment:
Hmm maybe we can address this when we go to enable the upgrade test (it is currently disabled) but by running the tests sequentially, PR pipeline will take a long time. Why exactly can we not run in parallel? Because the same region will be selected for both tests and it will fail?

Reply:
Correct. If run in parallel there is good chance the same region will be selected, and the limit for cloud connections is only 2 (so if both tests use a region having 1 existing connection, one of them will fail). I could adjust the limit for region selection to return only zones that have ZERO, which means even if they run in parallel and select the same one they should both be ok?

Reply:
I was thinking could we use a simple sleep but still run in parallel, but with a delay. But your solution sounds better :)

Reply:
Looks like this CI run took 23m, will need more runs though to get an average.